tests/accelerators/test_cpu.py

```diff
@@ -50,3 +51,97 @@ def setup_optimizers_in_pre_dispatch(self) -> bool:
     model = TestModel()
     trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, plugins=CustomPlugin(device=torch.device("cpu")))
     trainer.fit(model)
+
+
+def test_accelerator_on_reset_dataloader_hooks(tmpdir):
+    """
+    Ensures data-loader hooks are called at the right stages when using an accelerator.
+    """
+
+    class CustomAccelerator(CPUAccelerator):
+        mock_count: int = 0
+
+        def on_reset_train_dataloader(self, dataloader):
+            self.mock_count += 1
+            assert self.lightning_module.trainer.training
+            return super().on_reset_train_dataloader(dataloader)
+
+        def on_reset_val_dataloader(self, dataloader):
+            self.mock_count += 1
+            assert self.lightning_module.trainer.training or self.lightning_module.trainer.validating
+            return super().on_reset_val_dataloader(dataloader)
+
+        def on_reset_test_dataloader(self, dataloader):
+            self.mock_count += 1
+            assert self.lightning_module.trainer.testing
+            return super().on_reset_test_dataloader(dataloader)
+
+        def on_reset_predict_dataloader(self, dataloader):
+            self.mock_count += 1
+            assert self.lightning_module.trainer.predicting
+            return super().on_reset_predict_dataloader(dataloader)
+
+    model = BoringModel()
+    accelerator = CustomAccelerator(PrecisionPlugin(), SingleDevicePlugin(device=torch.device('cpu')))
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, accelerator=accelerator)
+    trainer.fit(model)
+    trainer.validate(model)
+    trainer.test(model)
+    trainer.predict(model, dataloaders=model.test_dataloader())
+    # assert that all loader hooks were called
+    assert accelerator.mock_count == 4
+
+    accelerator = CustomAccelerator(PrecisionPlugin(), SingleDevicePlugin(device=torch.device('cpu')))
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, accelerator=accelerator)
+    trainer.validate(model)
+    trainer.test(model)
+    trainer.predict(model)
+    # assert val/test/predict loader hooks were called
+    assert accelerator.mock_count == 3
+
+
+def test_plugin_on_reset_dataloader_hooks(tmpdir):
+    """
+    Ensures data-loader hooks are called at the right stages when using a plugin.
```

Comment:
ok it doesn't really do that it just counts any of the four possible hook calls ^^

Reply:
I'll make the test slightly better by increasing granularity!

Reply:
It's still not perfect, but now atleast we have a per stage count to ensure we're more explicit